packages/avatar/src/utils/getInitials/getInitials.ts

```diff
@@ -0,0 +1,87 @@
+import { supportedLatinRegex } from './latinRegex';
+
+interface Initials {
+  initials: string | null;
+  givenInitial: string | null;
+  surnameInitial: string | null;
+}
+
+const nullResult = {
+  initials: null,
+  givenInitial: null,
+  surnameInitial: null,
+};
+
+/**
+ * Returns the initials of the provided name(s)
+ *
+ * If only a single argument is provided,
+ * this string will be assumed to be the full name.
+ *
+ * Names including characters other than the English alphabet,
+ * and common European accented letters will be ignored,
+ * and will return `null`.
+ *
+ * **Known issue**: Names with suffixes (e.g. Jr., Sr., etc.)
+ * will include the first letter of the suffix in the surname initial
+ */
+export const getInitials = (
+  fullOrGivenName?: string,
+  surname?: string,
+): Initials => {
+  if (
+    (!fullOrGivenName && !!surname) || // no args provided
+    (fullOrGivenName && !supportedLatinRegex.test(fullOrGivenName)) || // arg0 is non-latin
+    (surname && !supportedLatinRegex.test(surname)) // arg1 is non-latin
+  ) {
+    return nullResult;
+  }
+
+  let givenInitial: string | null = null;
+  let surnameInitial: string | null = null;
+
+  if (surname) {
+    const givenName = fullOrGivenName;
+    givenInitial = givenName ? getSimpleInitials(givenName) : null;
+    surnameInitial = getSimpleInitials(surname);
+    const initials = `${givenInitial || ''}${surnameInitial || ''}`;
+
+    return {
+      initials,
+      givenInitial,
+      surnameInitial,
+    };
+  } else {
+    if (!fullOrGivenName) {
+      return nullResult;
+    }
+
+    const fullNameSegments = fullOrGivenName.split(' ');
+
+    const givenName = fullNameSegments[0];
+    givenInitial = getSimpleInitials(givenName);
+    surnameInitial = getSimpleInitials(fullNameSegments.slice(1).join(' '));
+  }
+
+  const initials = `${givenInitial || ''}${surnameInitial || ''}`;
+
+  return {
+    initials,
+    givenInitial,
+    surnameInitial,
+  };
+};
+
+const getSimpleInitials = (str: string): string => {
+  return str
+    .split(' ')
+    .map(n => n[0])
+    .filter(isUpperCase)
```

Comment:
Is it important to note that the string will return blank if the first letter of each name is not capitalized?

Reply:
Yeah that might be worth noting 👍